crates/sdk/src/default_main.rs

```diff
@@ -404,6 +415,53 @@ async fn post_query<C: Connector>(
     C::query(state.configuration(), state.state().await?, request).await
 }
 
+async fn print_schema_and_capabilities<Setup>(
+    setup: Setup,
+    command: PrintSchemaAndCapabilitiesCommand,
+) -> Result<()>
+where
+    Setup: ConnectorSetup,
+    Setup::Connector: Connector + 'static,
+    <Setup::Connector as Connector>::Configuration: Clone,
+    <Setup::Connector as Connector>::State: Clone,
+{
+    let server_state = init_server_state(setup, &command.configuration).await?;
+
+    let schema = Setup::Connector::get_schema(server_state.configuration()).await?;
+    let capabilities = get_capabilities::<Setup::Connector>().await;
+
+    print_json_schema_and_capabilities(schema, capabilities)?;
+
+    Ok(())
+}
+
+/// This foulness manually writes out a JSON object with schema and capabilities properties.
+/// We do it like this to avoid having to deserialize and reserialize any
+/// JsonResponse::Serialized values.
+fn print_json_schema_and_capabilities(
```

Comment:
Is it worth a test that runs this on on the example connector and makes sure that what is generated it at least parseable JSON?

Reply:
Good idea.

Reply:
@danieljharvey Done.